### ISSUE TITLE
`showNotification` doesn't support data-* attributes

### ISSUE THREAD

**Issue by OrRosenblatt** [mantinedev/mantine#5640]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.4.2

### What package has an issue?

@mantine/notifications

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Not applicable – issue is not related to the browser

### Describe the bug

When passing data-* attributes to `showNotification` or `notifications.show` functions a TS error is thrown:
```
Argument of type '{ message: string; "data-<attribute>": string; }' is not assignable to parameter of type 'NotificationData'.
  Object literal may only specify known properties, and '"data-<attribute>"' does not exist in type 'NotificationData'.
```

Example:
<img width="700" alt="data attribute used in show notification caused TS error" src="https://github.com/mantinedev/mantine/assets/20166480/8cb118bf-2e48-4130-8ca5-aef24af61935">

See link below for example code.

### If possible, include a link to a codesandbox with a minimal reproduction
https://codesandbox.io/p/sandbox/focused-moon-m85ccq

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
Fixed in [7.5.2](https://github.com/mantinedev/mantine/releases/tag/7.5.2)

**Comment by twiddler** [comment]
[This change](https://github.com/mantinedev/mantine/commit/e78a9bb9d23654349cf9413674dce08051322762) breaks this (simplified) code:

```ts
import { NotificationData, notifications } from '@mantine/notifications'

type OneTimeNotificationProps = Omit<NotificationData, 'color'>

function showSuccessNotification(props: OneTimeNotificationProps) {
    notifications.show({ // Argument of type '{ color: "green"; icon: JSX.Element; }' is not assignable to parameter of type 'NotificationData'. Property 'message' is missing in type '{ color: "green"; icon: JSX.Element; }' but required in type 'NotificationData'. ts(2345)
        color: 'green',
        ...props,
    })
}
```

TypeScript can no longer infer `OneTimeNotificationProps` narrowly:

```ts
type OneTimeNotificationProps = {
    [x: string]: any;
    [x: number]: any;
}
```
